spec/requests/post_spec.rb

```diff
@@ -33,6 +33,43 @@
     end
   end
 
+  describe 'GET index' do
+    context 'when user is authenticated' do
+      let!(:user) { create(:user) }
+      let!(:post1) { create(:post) }
+      let!(:post2) { create(:post) }
+
+      before do
+        sign_in user
+      end
+
+      it 'returns HTTP success' do
+        get root_path
+        expect(response).to be_successful
+      end
+
+      it 'shows created posts' do
+        get root_path
+        expect(response.body).to include(post1.body)
+        expect(response.body).to include(post2.body)
```

Comment:
To test this with a list of posts, you can maybe iterate over the variable of the list of posts.